### DISCUSSION TITLE
Blog component that features X number of the newest posts

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [gatsbyjs/gatsby#34271] [Ideas / Feature Requests] [answered]
I built a blog list component that I can call on any page. Right now it shows ALL my blog posts ,which is not a problem because I only have five. However, in a few weeks I may have a few hundred. So, I need to figure out how to only feature say the three newest ones. 

Query code [here](https://github.com/Isaac-Tait/Royal-Ridges-Retreat/blob/main/src/components/blogPostQuery.js)
Blog component code [here](https://github.com/Isaac-Tait/Royal-Ridges-Retreat/blob/main/src/components/blogPostList.js)

Anyone have suggestions (or maybe even an example) on how to make a component that only shows three (or any number really) of the most recent blog posts? Are there any plugins that would do that for me or do I need to code it all from scratch?

**Comment by violy** [comment, marked as answer]
I've not your graphQL query, but you  should [look at `limit` field argument](https://www.gatsbyjs.com/docs/graphql-reference/#limit).

**Reply by Isaac-Tait** [reply]
Oh my gosh! I had no idea that that even existed as an option AND it was simple to implement too ⭐  💯 Thank you very much 😄
